### PR TITLE
KOGITO-1866 Add Red Hat headers to all sources in Kogito Examples

### PR DESCRIPTION
**Pull request by bxf12315**

### CODE REVIEW

**Comment by bxf12315** [comment]
@cristianonicolai new PR , please review it.

File: .mvn/wrapper/MavenWrapperDownloader.java [hunk 1]
```diff
@@ -1,17 +1,17 @@
-/*
- * Copyright 2007-present the original author or authors.
```
Comment (ricardozanini) [inline .mvn/wrapper/MavenWrapperDownloader.java]: It's possible to keep this header? I don't think we should overwrite it.

**Review by ricardozanini — CHANGES_REQUESTED** [review]
Looks good, thanks for this effort. Just one question though: if I add a new java class to an example, once I run `mvn package` is the header added automatically by the plugin?

File: .mvn/wrapper/MavenWrapperDownloader.java (hunk 1 above)
Comment (bxf12315) [reply .mvn/wrapper/MavenWrapperDownloader.java]: updated

**Comment by bxf12315** [comment]
> Looks good, thanks for this effort. Just one question though: if I add a new java class to an example, once I run `mvn package` is the header added automatically by the plugin?

no, mvn license:format.

File: pom.xml [hunk 2]
```diff
@@ -383,6 +383,26 @@
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-source-plugin</artifactId>
       </plugin>
+      <plugin>
+        <groupId>com.mycila</groupId>
+        <artifactId>license-maven-plugin</artifactId>
```
Comment (cristianonicolai) [inline pom.xml:377]: @bxf12315 please use checkstyle instead, see https://github.com/kiegroup/jbpm-wb/blob/master/pom.xml#L165

Comment (bxf12315) [reply pom.xml:377]: if use checkstyle instead it how to automatically add license header? 
or both? license-maven-plugin also has to check feature base on https://github.com/kiegroup/kogito-examples/pull/230/files#diff-9556455f12108bc30ce288613f3428da. style of checkstyle plugin different to default of license-maven-plugin, so we can not use them.

**Review by evacchi — APPROVED** [review]
looks good. Double checking: you haven't added license header to Template files right? that's correct (otherwise the license would end up in generated code)

**Comment by bxf12315** [comment]
> looks good. Double checking: you haven't added license header to Template files right? that's correct (otherwise the license would end up in generated code)

Thanks, I am not sure what you mean, it has a template file for license: https://github.com/kiegroup/kogito-examples/pull/230/files#diff-9556455f12108bc30ce288613f3428da.I use a mavn plugin:https://github.com/mycila/license-maven-plugin

**Comment by ricardozanini** [comment]
> > looks good. Double checking: you haven't added license header to Template files right? that's correct (otherwise the license would end up in generated code)
> 
> Thanks, I am not sure what you mean, it has a template file for license: https://github.com/kiegroup/kogito-examples/pull/230/files#diff-9556455f12108bc30ce288613f3428da.I use a mavn plugin:https://github.com/mycila/license-maven-plugin

@bxf12315 what he meant is that we have some java files that serve as a template to generate code for custom Kogito services. Those files should NOT contain the headers or will end up being on users' code.

EDIT: @evacchi do we have template files on examples? :thinking:

**Review by ricardozanini — APPROVED** [review]

**Comment by evacchi** [comment]
@ricardozanini no we don't :D sorry got confused

**Comment by cristianonicolai** [comment]
@bxf12315 can you check why the build is failing

**Comment by ricardozanini** [comment]
> @bxf12315 can you check why the build is failing

Looks like a problem in the license header plugin configuration:

```
Failed to execute goal com.mycila:license-maven-plugin:3.0:check (default) on project kogito-examples: Some files do not have the expected license header -> [Help 1]
```

**Comment by bxf12315** [comment]
> @bxf12315 can you check why the build is failing

I find many warning is 
[WARNING] Missing header in: /home/jenkins/workspace/IE_kogito_kogito-examples_PR-230/kogito-runtimes/drools/drools-core/src/main/java/org/drools/core/fluent/impl/ExecutableBuilderImpl.java
[WARNING] Missing header in: /home/jenkins/workspace/IE_kogito_kogito-examples_PR-230/kogito-runtimes/drools/drools-core/src/main/java/org/drools/core/fluent/impl/CommandRegister.java
[WARNING] Missing header in: /home/jenkins/workspace/IE_kogito_kogito-examples_PR-230/kogito-runtimes/drools/drools-core/src/main/java/org/drools/core/fluent/impl/BatchImpl.java
[WARNING] Missing header in: /home/jenkins/workspace/IE_kogito_kogito-examples_PR-230/kogito-runtimes/drools/drools-core/src/main/java/org/drools/core/fluent/impl/Batch.java
[WARNING] Missing header in: /home/jenkins/workspace/IE_kogito_kogito-examples_PR-230/kogito-runtimes/drools/drools-core/src/main/java/org/drools/core/fluent/impl/BaseBatchWithProcessFluent.java
[WARNING] Missing header in: /home/jenkins/workspace/IE_kogito_kogito-examples_PR-230/kogito-runtimes/drools/drools-core/src/main/java/org/drools/core/fluent/impl/BaseBatchFluent.java.
I do not know kogito jenkins build how to build all repo using examples's pom.
I add two exclude rules, hope it will be work.

**Comment by bxf12315** [comment]
@cristianonicolai I find a code line[1] in Jenkins file.
when jenkins build kogito-examples,it do to go dir kogito-examples. Is it special settings? If it's not , I will change it.
[1] https://github.com/kiegroup/kogito-examples/blob/54eef215666bb35a4a72d222dadd76f4c149e719/Jenkinsfile#L43

**Comment by cristianonicolai** [comment]
> @cristianonicolai I find a code line[1] in Jenkins file.
> when jenkins build kogito-examples,it do to go dir kogito-examples. Is it special settings? If it's not , I will change it.
> [1]
> 
> https://github.com/kiegroup/kogito-examples/blob/54eef215666bb35a4a72d222dadd76f4c149e719/Jenkinsfile#L43

@bxf12315 build is fine as it should build all related modules, it seems that your maven check that is validating classes that are not part of this repository.

File: pom.xml (hunk 2 above)
Comment (cristianonicolai) [reply pom.xml:377]: style can be configured, see https://github.com/kiegroup/jbpm-wb/blob/master/pom.xml#L177

**Comment by bxf12315** [comment]
> > @cristianonicolai I find a code line[1] in Jenkins file.
> > when jenkins build kogito-examples,it do to go dir kogito-examples. Is it special settings? If it's not , I will change it.
> > [1]
> > https://github.com/kiegroup/kogito-examples/blob/54eef215666bb35a4a72d222dadd76f4c149e719/Jenkinsfile#L43
> 
> @bxf12315 build is fine as it should build all related modules, it seems that your maven check that is validating classes that are not part of this repository.

"stage('Build kogito-examples') {
            steps {
                script {
                    maven.runMavenWithSubmarineSettings('clean install', false)
                }"
The build first step is build kogito-runtimes.second is kogito-apps, the two steps all go to them-slef's folder,but build kogito-exapmles do not have dir,
I think the jenkins build's localtion is root.so It will check all folder.

File: pom.xml (hunk 2 above)
Comment (bxf12315) [reply pom.xml:377]: yes I know it. but if we remove the plugin,how to automatically add license header? and it has check feature.,

**Review by evacchi — APPROVED** [review]
needs rebase :(

**Comment by bxf12315** [comment]
> needs rebase :(

done

**Review by cristianonicolai — APPROVED** [review]

File: pom.xml [hunk 3]
```diff
@@ -390,6 +390,28 @@
...
```
Comment (cristianonicolai) [inline pom.xml:377]: @bxf12315 can you add a specific version and manage it via pluginmanagement section please.

File: licensesheader.txt [hunk 4]
```diff
@@ -0,0 +1,13 @@
+Copyright 2020 Red Hat, Inc. and/or its affiliates.
```
Comment (cristianonicolai) [inline licensesheader.txt]: @bxf12315 what happens when the year changes? :)

**Review by cristianonicolai — CHANGES_REQUESTED** [review]

File: licensesheader.txt (hunk 4 above)
Comment (bxf12315) [reply licensesheader.txt]: when year changes the template need change.

Comment (cristianonicolai) [reply licensesheader.txt]: > variable replacement: You can add some variable in your header, such as ${year}, ${owner} and they will be replaced by the corresponding values taken from the pom or system properties.

From the plugin docs

Comment (bxf12315) [reply licensesheader.txt]: yes, the way also need change the template and pom.xml.
Like this 

properties
 year 2020year
properties

File: pom.xml (hunk 3 above)
Comment (bxf12315) [reply pom.xml:377]: done

Comment (MarianMacik) [reply pom.xml:377]: Can you specify it in accordance with convention? <version.license.plugin>? Thank you.

@cristianonicolai does kogito-examples not depend on kogito-runtimes BOM on purpose? Maybe we could centralize this even more (as a part of another PR).

Comment (cristianonicolai) [reply pom.xml:377]: @MarianMacik yeah we should do that, see https://issues.redhat.com/browse/KOGITO-1371

Comment (bxf12315) [reply pom.xml:377]: done

**Comment by evacchi** [comment]
some files are still missing headers

**Comment by bxf12315** [comment]
> some files are still missing headers
which java file miss license header?
 New updated, I use license-maven-plugin-git, it needs rum mvn license:check firstly.it pass by license:check.

**Review by cristianonicolai — APPROVED** [review]

**Review by MarianMacik — APPROVED** [review]

**Review by radtriste — APPROVED** [review]

**Pull request merged**